app/services/permissions/order.rb

```diff
@@ -35,6 +39,12 @@ def editable_line_items
 
     private
 
+    attr_reader :search_params
+
+    def search_orders?
+      search_params.present?
+    end
```

Comment:
I'm not sure with ransack but maybe this could be simpler. I'm imagining using this method in the cases above:

```suggestion
    def filtered_orders
      if search_params.present?
        Spree::Order.complete.not_state(:canceled).search(search_params).result
      else
        Spree::Order.scoped
      end
    end
```

```rb
    def editable_orders
      filtered_orders.where(
        managed_orders_where_values.
          or(coordinated_orders_where_values)
      )
    end
```

Reply:
I tried making these _exact_ changes and there were some issues with the specs; the returned results were not correct. 

I've added a new commit where it's refactored in a fairly similar way, adding a #filtered_orders method. Also renamed #search_orders? to #filter_orders?